bundles/org.openhab.binding.lghombot/src/main/java/org/openhab/binding/lghombot/internal/LGHomBotHandler.java

```diff
@@ -0,0 +1,722 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lghombot.internal;
+
+import static org.openhab.binding.lghombot.internal.LGHomBotBindingConstants.*;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.imageio.ImageIO;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.util.UrlEncoded;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LGHomBotHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Fredrik Ahlström - Initial contribution
+ */
+@NonNullByDefault
+public class LGHomBotHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LGHomBotHandler.class);
+
+    // This is setup in initialize().
+    private LGHomBotConfiguration config = new LGHomBotConfiguration();
+
+    private @Nullable ScheduledFuture<?> refreshTimer;
+
+    // State of HomBot
```

Comment:
You could even consider not having this state at all and (only) schedule an extra refresh which also updates all the states anyway.  It saves a lot of code.

Reply:
The reason why I wanted to have a local state was because of the polling and that it gets all states each time.

Reply:
It makes sense, by storing the state you can immediately deliver the new data if an extra channel is linked. So it has benefits.

The idea of the refresh command is to trigger the binding to fetch new data (if possible quick) it does not require you to give (in this case stale) data right away. 

You could also (as an example) cancel the current polling job and schedule one with a delay of 0.

Some bindings use the ExpiringCacheAsync to do this, it will fetch new data once it expires but does not refresh if the data is still recent. 

For me it's NOT blocking, it was mainly that class became very big with a lot of related fields that were not closely related to the handler itself.